### ISSUE TITLE
[0.75.x/0.76.x] Failing to build on Windows - Could not move temporary workspace - Error resolving plugin [id: 'com.facebook.react.settings'] - java.io.UncheckedIOException

### ISSUE THREAD

**Issue by adhikarisandil32** [facebook/react-native#46210]
**IMPORTANT: The latest official update on this issue is here** https://github.com/facebook/react-native/issues/46210#issuecomment-2346835978

```
FAILURE: Build failed with an exception.

* Where:
Settings file 'F:\tutorial-practice\frontend-development\react-native\native-cli-todo-app-2\android\settings.gradle' line: 2

* What went wrong:
Error resolving plugin [id: 'com.facebook.react.settings']
> java.io.UncheckedIOException: Could not move temporary workspace (F:\tutorial-practice\frontend-development\react-native\native-cli-todo-app-2\android\.gradle\8.8\dependencies-accessors\569c8b261a8a714d7731d5f568e0e5c05babae10-29ae1a11-17a6-474d-825c-7114fe1603d2) to immutable location (F:\tutorial-practice\frontend-development\react-native\native-cli-todo-app-2\android\.gradle\8.8\dependencies-accessors\569c8b261a8a714d7731d5f568e0e5c05babae10)
```

This is the full error. The command I've used to create the project is `npx @react-native-community/cli init TodoApp2 --directory ./native-cli-todo-app-2`. The projects is created successfully. After I run `npm run start` and hit `a`, for android, build fails showing the above error. Another project which uses 0.74.3 is working just fine. Creating new project automatically uses 0.75.2 version which is fine, but it shows the error while building because of which development server doesn't start.
![Screenshot 2024-08-21 010605](https://github.com/user-attachments/assets/15d3e0bb-760f-4d17-b888-31642e633297)

**Comment by Ibeenoch** [comment]
still can't find a fix

**Comment by raahimkhan** [comment]
Facing same issue

**Comment by raahimkhan** [comment]
> > Downgrading to RN `0.74.5` and Gradle `8.5` solves the issue. Wonder how the new release has been working for other folks. May just be an issue with those with simulated devices running on emulators?
> 
> @Agent009 I use it on real actual device and I face the same issue. I don't know. Nobody seems to be working on it. And I don't know react native enough to work on it. I don't know how to direct contributers to this issue.

I have the same versions but still getting the error.

**Comment by ralppppy** [comment]
This works for me https://github.com/facebook/react-native/issues/46046#issuecomment-2295388352

**Comment by GangulyYadavCDAT** [comment]
i was facing the same issue what i just did to solve is 
`cd android`
then 
`gradlew` clean or `./gradlew clean`
then 
`gradlew` build or `./gradlew build `
whatever works for you give it a try 
then run your project

**Comment by cortinico** [comment]
Due to a various of random comments, it's extremely hard for us to understand what happened here.

**Please refrain from commenting "same issue" as that creates only unnecessary noise**

As far as I understood:

1. The issue is happening **only** on Windows
2. The issue is happening **only** on React Native 0.75.x

If that is NOT the case for you -> Open another issue and **do not comment here**.

I've sharing some steps that could (hopefully help till we try to identify what's the issue):

1. Make sure you don't have a version of `@react-native/gradle-plugin` specified in your package.json. If you have it, please remove it.
2. Check with `yarn why @react-native/gradle-plugin` that the output is:
```bash 
$ yarn why @react-native/gradle-plugin
├─ react-native@npm:0.75.1
│  └─ @react-native/gradle-plugin@npm:0.75.1 (via npm:0.75.1)
│
└─ react-native@npm:0.75.1 [473c9]
   └─ @react-native/gradle-plugin@npm:0.75.1 (via npm:0.75.1)
```
3. Kill all the Gradle daemons with `cd android && ./gradlew --stop`
4. Remove the `.gradle` folder with `cd android && rm -rf .gradle/ && rm -rf ~/.gradle/caches`
5. Attemp to downgrade to Gradle 8.7 by changing this file `android/gradle/wrapper/gradle-wrapper.properties`:

```diff
distributionBase=GRADLE_USER_HOME
distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-8.8-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-8.7-all.zip
networkTimeout=10000
validateDistributionUrl=true
zipStoreBase=GRADLE_USER_HOME
zipStorePath=wrapper/dists
```

6. Attempt to upgrade to Gradle 8.10 by editing the same file.

Please report back if this solves the issue @Agent009

**Comment by cortinico** [comment]
(Related to https://github.com/gradle/gradle/issues/27844)

**Comment by MDavidMR** [comment]
> 2\. yarn why @react-native/gradle-plugin

Hi! I tried your suggestions but the problem persists...

I'm so confused because yesterday I created a new proyect with 0.75.2 but it didn't run with 'npm run andoid' so I changed it version to

**Comment by MDavidMR** [comment]
> Due to a various of random comments, it's extremely hard for us to understand what happened here.
> 
> **Please refrain from commenting "same issue" as that creates only unnecessary noise**
> 
> As far as I understood:
> 
> 1. The issue is happening **only** on Windows
> 2. The issue is happening **only** on React Native 0.75.x
> 
> If that is NOT the case for you -> Open another issue and **do not comment here**.
> 
> I've sharing some steps that could (hopefully help till we try to identify what's the issue):
> 
> 1. Make sure you don't have a version of `@react-native/gradle-plugin` specified in your package.json. If you have it, please remove it.
> 2. Check with `yarn why @react-native/gradle-plugin` that the output is:
> 
> ```shell
> $ yarn why @react-native/gradle-plugin
> ├─ react-native@npm:0.75.1
> │  └─ @react-native/gradle-plugin@npm:0.75.1 (via npm:0.75.1)
> │
> └─ react-native@npm:0.75.1 [473c9]
>    └─ @react-native/gradle-plugin@npm:0.75.1 (via npm:0.75.1)
> ```
> 
> 3. Kill all the Gradle daemons with `cd android && ./gradlew --stop`
> 4. Remove the `.gradle` folder with `cd android && rm -rf .gradle/ && rm -rf ~/.gradle/caches`
> 5. Attemp to downgrade to Gradle 8.7 by changing this file `android/gradle/wrapper/gradle-wrapper.properties`:
> 
> ```diff
> distributionBase=GRADLE_USER_HOME
> distributionPath=wrapper/dists
> -distributionUrl=https\://services.gradle.org/distributions/gradle-8.8-all.zip
> +distributionUrl=https\://services.gradle.org/distributions/gradle-8.7-all.zip
> networkTimeout=10000
> validateDistributionUrl=true
> zipStoreBase=GRADLE_USER_HOME
> zipStorePath=wrapper/dists
> ```
> 
> 6. Attempt to upgrade to Gradle 8.10 by editing the same file.
> 
> Please report back if this solves the issue @Agent009


2. yarn why @react-native/gradle-plugin

Hi cortinico, I tried your suggestions step by step but the problem persists...

I'm so confused because yesterday I created a new proyect with 0.75.2 but it didn't run with 'npm run andoid' ( 'com.facebook.react.settings' error) so I changed it version to 0.75.1, after this, the project worked.

Today I tried to create other project following the previous 'hack' but 'com.facebook.react.settings' error is showing again.

**Comment by cipolleschi** [comment]
Hi @MDavidMR can you try to run `gradlew clean` and see if it resets the state of the machine?
If you keep 0.75.2 and downgrade gradle to 8.7 as shown by @cortinico above, is the problem fixed?

**Comment by harshmdr-devslane** [comment]
Not working for version with RN 0.75.1.

**Comment by Cyberfolks** [comment]
> Hi @MDavidMR can you try to run `gradlew clean` and see if it resets the state of the machine? If you keep 0.75.2 and downgrade gradle to 8.7 as shown by @cortinico above, is the problem fixed?

yes it work for me i just downgraded gradle, deleted app/buld folder and then try to clean gradle. its done

**Comment by projectninjatech** [comment]
cd android
./gradlew clean
./gradlew --stop
Then, go back to the root directory of your project and run:
npx react-native run-android

I did this and it is working for me.

**Comment by VcRajaa** [comment]
> cd android ./gradlew clean ./gradlew --stop Then, go back to the root directory of your project and run: npx react-native run-android
> 
> I did this and it is working for me. You don't have to downgrade gradle with this method.

I followed this, and working fine for me. Thanks dear

**Comment by Renmsd** [comment]
> Downgrading to RN `0.74.5` and Gradle `8.5` solves the issue. Wonder how the new release has been working for other folks. May just be an issue with those with simulated devices running on emulators?
 
how did you downgrade react native?

**Comment by cipolleschi** [comment]
@Renmsd don't do it, let's try first to run `./gradlew clean && ./gradlew --stop` and see if it fixes

**Comment by Renmsd** [comment]
> ./gradlew clean && ./gradlew --stop

neglect that the error msg has changed (dont know why) but there is no such file with the name "./gradlew "

**Comment by MDavidMR** [comment]
> Hi @MDavidMR can you try to run `gradlew clean` and see if it resets the state of the machine? If you keep 0.75.2 and downgrade gradle to 8.7 as shown by @cortinico above, is the problem fixed?

Finally a working solution, no need to downgrade Gradle to 8.7. I'm just starting out with React-Native development and I was finding it very frustrating not being able to run a project immediately after creating it.

I also executed the command suggested by @projectninjatech : `./gradlew --stop`

Thank you very much

**Comment by cipolleschi** [comment]
@Renmsd you have it right above the settings file....

You need to run `./gradlew` from the android folder:
```
cd <path-to-your-project>\android
./gradlew clean
./gradlew --stop
```

perhaps on windows the command is slightly different (perhaps it is `./gradlew.bat`?), but it's there

![363226709-8cd22d47-5d34-474c-81fb-a9aa6f48786b](https://github.com/user-attachments/assets/2c45484b-054c-4427-a271-5650a456761b)

**Comment by Agent009** [comment]
> 2\. yarn why @react-native/gradle-plugin

Hi,

I tried both gradle 8.7 as well as 8.10 following your steps:

- Stop gradle daemons
- 
- Change gradle version
- Delete `android/.gradle`, `android/build` and `~/.gradle/caches`

Gradle 8.7 gets stuck at the same point when running `./gradlew clean`, stopping at this point in the gradle daemon output log:

````
2024-08-31T12:32:59.128+0100 [DEBUG] [org.gradle.launcher.daemon.server.exec.ExecuteBuild] Executing build with daemon context: DefaultDaemonContext[uid=11009b0a-f51f-48fd-abb3-36db1a89378b,javaHome=C:\Program Files\Java\jdk-21,daemonRegistryDir=C:\Users\user\.gradle\daemon,pid=28388,idleTimeout=10800000,priority=NORMAL,applyInstrumentationAgent=true,daemonOpts=-XX:MaxMetaspaceSize=512m,--add-opens=java.base/java.util=ALL-UNNAMED,--add-opens=java.base/java.lang=ALL-UNNAMED,--add-opens=java.base/java.lang.invoke=ALL-UNNAMED,--add-opens=java.prefs/java.util.prefs=ALL-UNNAMED,--add-opens=java.base/java.nio.charset=ALL-UNNAMED,--add-opens=java.base/java.net=ALL-UNNAMED,--add-opens=java.base/java.util.concurrent.atomic=ALL-UNNAMED,-Xmx2048m,-Dfile.encoding=UTF-8,-Duser.country=GB,-Duser.language=en,-Duser.variant]
````

Running `npm run android` ends with the same error as before:

````
Error resolving plugin [id: 'com.facebook.react.settings']
> java.io.UncheckedIOException: Could not move temporary workspace (C:\project\android\.gradle\8.7\dependencies-accessors\569c8b26
1a8a714d7731d5f568e0e5c05babae10-19d4b8ef-c0fc-408d-b528-17bd9053d50c) to immutable location (C:\project\android\.gradle\8.7\dependencies-accessors\569c8b261a8a714d7731d5f568e0e5c05babae10)
````

and gradle 8.10 errors with the following:

````
e: file:///project/node_modules/@react-native/gradle-plugin/react-native-gradle-plugin/build.gradle.kts:10:49: Unresolved reference: serviceOf
e: file:///project/node_modules/@react-native/gradle-plugin/react-native-gradle-plugin/build.gradle.kts:58:11: Unresolved reference: serviceOf

FAILURE: Build completed with 2 failures.

1: Task failed with an exception.
-----------
* Where:
Build file 'C:\project\node_modules\@react-native\gradle-plugin\react-native-gradle-plugin\build.gradle.kts' line: 10

* What went wrong:
Script compilation errors:

  Line 10: import org.gradle.configurationcache.extensions.serviceOf
                                                           ^ Unresolved reference: serviceOf

  Line 58:           serviceOf<ModuleRegistry>()
                     ^ Unresolved reference: serviceOf
````

**Comment by Vantiboolean** [comment]
When I downgraded the yarn version to 1.22, it worked

**Comment by deeeed** [comment]
This issue also happens on macos

**Comment by shubhkk07** [comment]
> ```
> FAILURE: Build failed with an exception.
> 
> * Where:
> Settings file 'F:\tutorial-practice\frontend-development\react-native\native-cli-todo-app-2\android\settings.gradle' line: 2
> 
> * What went wrong:
> Error resolving plugin [id: 'com.facebook.react.settings']
> > java.io.UncheckedIOException: Could not move temporary workspace (F:\tutorial-practice\frontend-development\react-native\native-cli-todo-app-2\android\.gradle\8.8\dependencies-accessors\569c8b261a8a714d7731d5f568e0e5c05babae10-29ae1a11-17a6-474d-825c-7114fe1603d2) to immutable location (F:\tutorial-practice\frontend-development\react-native\native-cli-todo-app-2\android\.gradle\8.8\dependencies-accessors\569c8b261a8a714d7731d5f568e0e5c05babae10)
> ```
> 
> This is the full error. The command I've used to create the project is `npx @react-native-community/cli init TodoApp2 --directory ./native-cli-todo-app-2`. The projects is created successfully. After I run `npm run start` and hit `a`, for android, build fails showing the above error. Another project which uses 0.74.3 is working just fine. Creating new project automatically uses 0.75.2 version which is fine, but it shows the error while building because of which development server doesn't start. ![Screenshot 2024-08-21 010605](https://private-user-images.githubusercontent.com/94738431/359618830-15d3e0bb-760f-4d17-b888-31642e633297.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjUyNjYwMjgsIm5iZiI6MTcyNTI2NTcyOCwicGF0aCI6Ii85NDczODQzMS8zNTk2MTg4MzAtMTVkM2UwYmItNzYwZi00ZDE3LWI4ODgtMzE2NDJlNjMzMjk3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MDIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTAyVDA4Mjg0OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY3OGRmMTk4OTRiODNhNjg4ZmI2NjU4ZTBmMjYxYmNmZjdjNzc4MWQ0ZjU2YTAwZWY0ZTgwYzQ2MDNiMGY0MDImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.EDmccawlYiMdA8WVm-4EhKg6i7U_wVf4X_c8XSBQX_Y)

didn't work for Anyone can share other suggestions.

**Comment by Renmsd** [comment]
in Android folder I deleted `.gradel ` and then ran these two lines `rd /s /q .gradle` and `gradlew clean` and it worked ( windows) I used Gardle 8.7

**Comment by elmennani89** [comment]
I'm also facing this issue but in Mac with m1 chip. I used pnpm to install dependencies. When I checked `node_modules/@react-native` folder I couldn't find `gradle-plugin` inside it. so I installed it `pnpm add @react-native/gradle-plugin` and the issue went away. the issue I encounter now is that the app crashes. this a new project created with the cli.

**Comment by singhJasvinder101** [comment]
Was getting the same error after installing node_modules/ setup of react native project
close your terminal and create new one then run

`npm run android --clear-cache`

**Comment by kazisalman** [comment]
Delete the ./gradle folder in C drive , then hit the commands again, it usually happens due to difference in gradle versions ( java versions)

**Comment by UIT19521334** [comment]
still happen to me please help i try 
```
cd android && ./gradlew clean && ./gradlew --stop
```
but it still show error
`Where: Settings file 'D:\DAT\PoultryApps\android\settings.gradle' line: 2 * What went wrong: Error resolving plugin [id: 'com.facebook.react.settings'] > java.io.UncheckedIOException: Could not move temporary workspace`

**Comment by MalekZishan** [comment]
> Due to a various of random comments, it's extremely hard for us to understand what happened here.
> 
> **Please refrain from commenting "same issue" as that creates only unnecessary noise**
> 
> As far as I understood:
> 
> 1. The issue is happening **only** on Windows
> 2. The issue is happening **only** on React Native 0.75.x
> 
> If that is NOT the case for you -> Open another issue and **do not comment here**.
> 
> I've sharing some steps that could (hopefully help till we try to identify what's the issue):
> 
> 1. Make sure you don't have a version of `@react-native/gradle-plugin` specified in your package.json. If you have it, please remove it.
> 2. Check with `yarn why @react-native/gradle-plugin` that the output is:
> 
> ```shell
> $ yarn why @react-native/gradle-plugin
> ├─ react-native@npm:0.75.1
> │  └─ @react-native/gradle-plugin@npm:0.75.1 (via npm:0.75.1)
> │
> └─ react-native@npm:0.75.1 [473c9]
>    └─ @react-native/gradle-plugin@npm:0.75.1 (via npm:0.75.1)
> ```
> 
> 3. Kill all the Gradle daemons with `cd android && ./gradlew --stop`
> 4. Remove the `.gradle` folder with `cd android && rm -rf .gradle/ && rm -rf ~/.gradle/caches`
> 5. Attemp to downgrade to Gradle 8.7 by changing this file `android/gradle/wrapper/gradle-wrapper.properties`:
> 
> ```diff
> distributionBase=GRADLE_USER_HOME
> distributionPath=wrapper/dists
> -distributionUrl=https\://services.gradle.org/distributions/gradle-8.8-all.zip
> +distributionUrl=https\://services.gradle.org/distributions/gradle-8.7-all.zip
> networkTimeout=10000
> validateDistributionUrl=true
> zipStoreBase=GRADLE_USER_HOME
> zipStorePath=wrapper/dists
> ```
> 
> 6. Attempt to upgrade to Gradle 8.10 by editing the same file.
> 
> Please report back if this solves the issue @Agent009



tnx bro save my day

**Comment by MuruganBalachandran** [comment]
> > facing same issue
> 
> @rghanks
> 
> I've found a trick though. hit `npm run android`, let it do its job all the way. Now exit the terminal, clear the console and then again hit `npm run start`. Work for me.



Well, It is Working for me also.
i have been finding solution , for this problem . Thank you @rghanks

**Comment by Coaquera** [comment]
I have tried several things, I managed to work with
“react-native: “0.75.2” and /gradle-8.8-all.zip

The order is important:
 1.- Create the project -> npx @react-native-community/cli@latest init AwesomeProject
 2.- If we have the virtual android we must close it.
 3.- click on start virtual android(I use android estudio) 
 4.- finally -> npm run android

**Comment by Anup481** [comment]
> ```
> FAILURE: Build failed with an exception.
> 
> * Where:
> Settings file 'F:\tutorial-practice\frontend-development\react-native\native-cli-todo-app-2\android\settings.gradle' line: 2
> 
> * What went wrong:
> Error resolving plugin [id: 'com.facebook.react.settings']
> > java.io.UncheckedIOException: Could not move temporary workspace (F:\tutorial-practice\frontend-development\react-native\native-cli-todo-app-2\android\.gradle\8.8\dependencies-accessors\569c8b261a8a714d7731d5f568e0e5c05babae10-29ae1a11-17a6-474d-825c-7114fe1603d2) to immutable location (F:\tutorial-practice\frontend-development\react-native\native-cli-todo-app-2\android\.gradle\8.8\dependencies-accessors\569c8b261a8a714d7731d5f568e0e5c05babae10)
> ```
> 
> This is the full error. The command I've used to create the project is `npx @react-native-community/cli init TodoApp2 --directory ./native-cli-todo-app-2`. The projects is created successfully. After I run `npm run start` and hit `a`, for android, build fails showing the above error. Another project which uses 0.74.3 is working just fine. Creating new project automatically uses 0.75.2 version which is fine, but it shows the error while building because of which development server doesn't start. ![Screenshot 2024-08-21 010605](https://private-user-images.githubusercontent.com/94738431/359618830-15d3e0bb-760f-4d17-b888-31642e633297.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU4NjYzMzAsIm5iZiI6MTcyNTg2NjAzMCwicGF0aCI6Ii85NDczODQzMS8zNTk2MTg4MzAtMTVkM2UwYmItNzYwZi00ZDE3LWI4ODgtMzE2NDJlNjMzMjk3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTA5VDA3MTM1MFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTdiNTMwMjM3NGU0ZmQyZmNiNTNmM2FkNDkzY2JlMGNlNjkyZGFlM2E4MjRmYmIzNTVjZTNmMjJkNTI5MzQ2NWImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.KQnANO68K-o9vwg4UDp3c0CRi9a8L7PlMVL-3I5cEzQ)

**Update to JDK 21 to solve this issue**

**Comment by websasindu** [comment]
This method works for me.

Close Everything.
Set Java Home To Java 17
npx @react-native-community/cli@latest init AwesomeProject
npx react-native run-android

and all errors gone!

**Comment by K0ffeeKat** [comment]
I was upgrading from 0.74.2 to 0.75.2 and got the same error on Android, here is what worked for me (using gradle-8.8-all.zip):

- remove node_modules, yarn.lock and android/.gradle
- yarn install
- cd android && ./gradlew --stop
- ./gradlew clean
- npx react-native start --reset-cache in the root of the project

Hope this will be helpful

**Comment by redskull-git** [comment]
> I was upgrading from 0.74.2 to 0.75.2 and got the same error on Android, here is what worked for me (using gradle-8.8-all.zip):
> 
> * remove node_modules, yarn.lock and android/.gradle
> * yarn install
> * cd android && ./gradlew --stop
> * ./gradlew clean
> * npx react-native start --reset-cache in the root of the project
> 
> Hope this will be helpful

It's work, thanks

**Comment by felimoles** [comment]
React native 0.75.1
jdk 17
gradle 8.7
Windows 11
`npm run start `
Error resolving plugin [id: 'com.facebook.react.settings'] , then  not close emulador and 
`npm run android`
work

 so weird

**Comment by TNT-Likely** [comment]
> cd android ./gradlew clean ./gradlew --stop Then, go back to the root directory of your project and run: npx react-native run-android
> 
> I did this and it is working for me. You don't have to downgrade gradle with this method.

it's work for me , thanks

**Comment by PURPLE77777** [comment]
Hi guys!
I resolved issue with "Error resolving plugin [id: 'com.facebook.react.settings']" my own way:
1) init a project npx @react-native-community/cli@latest init AwesomeProject
you can just try to type a command "npx react-native run-android" at least for downloading gradle 8.8, but we will get the expecting issue.
2) go to the file: "<YOUR_PROJECT>/node_modules/@react-native/gradle-plugin/react-native-gradle-plugin/src/main/kotlin/com/facebook/react/ReactPlugin.kt" and edit a line 97 like this:
from: val jvmVersion = Jvm.current()?.javaVersion?.majorVersion
to:     val jvmVersion = Jvm.current().javaVersion?.majorVersion
just remove the first question mark, it's important because if i don't do this i continue to get the issue
3) launch android studio and make a build (before this step you should to close emulator, stop gradle daemons if you launched before by commands "cd <YOUR_PROJECT>/android" and "./gradlew --stop", clear gradle cache by commmands "cd <YOUR_PROJECT>/android" and "./gradlew clean")
4) run the command "npx react-native run-android"

Env:
Win 11
React-native 0.75.3
gradle 8.8
jdk-17.0.11
Android Studio Koala | 2024.1.1 Patch 2

**Comment by lalithasj** [comment]
Hi,

React-Native@0.75.3 - build failed

BUILD FAILED in 17s
error Failed to install the app. Command failed with exit code 1: gradlew.bat app:installDebug -PreactNativeDevServerPort=8081 FAILURE: Build failed with an exception. * Where: Settings file 'E:\ReactNative\TestSept13\android\settings.gradle' line: 2 * What went wrong: Error resolving plugin [id: 'com.facebook.react.settings'] > java.io.UncheckedIOException: Could not move temporary workspace (E:\ReactNative\TestSept13\android\.gradle\8.8\dependencies-accessors\569c8b261a8a714d7731d5f568e0e5c05babae10-1ad873d5-53fc-465e-8179-54485ab46d5a) to immutable location (E:\ReactNative\TestSept13\android\.gradle\8.8\dependencies-accessors\569c8b261a8a714d7731d5f568e0e5c05babae10) * Try: > Run with --stacktrace option to get the stack trace. > Run with --info or --debug option to get more log output. > Run with --scan to get full insights. > Get more help at https://help.gradle.org. BUILD FAILED in 17s

Then,

remove node_modules, yarn.lock and android/.gradle
yarn install
cd android && ./gradlew --stop
./gradlew clean
./gradlew clean
Starting a Gradle Daemon, 2 stopped Daemons could not be reused, use --status for details

FAILURE: Build failed with an exception.

* Where:
Settings file 'E:\ReactNative\TestSept13\android\settings.gradle' line: 2

* What went wrong:
Error resolving plugin [id: 'com.facebook.react.settings']
> java.io.UncheckedIOException: Could not move temporary workspace (E:\ReactNative\TestSept13\android\.gradle\8.8\dependencies-accessors\569c8b261a8a714d7731d5f568e0e5c05babae10-dc75b5f3-542f-4313-895c-495df9b762cf) to immutable location (E:\ReactNative\TestSept13\android\.gradle\8.8\dependencies-accessors\569c8b261a8a714d7731d5f568e0e5c05babae10) 



npx react-native start --reset-cache in the root of the project

FAILURE: Build failed with an exception.

* Where:
Settings file 'E:\ReactNative\TestSept13\android\settings.gradle' line: 2

* What went wrong:
Error resolving plugin [id: 'com.facebook.react.settings']
> java.io.UncheckedIOException: Could not move temporary workspace (E:\ReactNative\TestSept13\android\.gradle\8.8\dependencies-accessors\569c8b261a8a714d7731d5f568e0e5c05babae10-4a79e70b-130f-4edb-b300-e277ff551d05) to immutable location (E:\ReactNative\TestSept13\android\.gradle\8.8\dependencies-accessors\569c8b261a8a714d7731d5f568e0e5c05babae10)      

Appreciate any suggestions

**Comment by Sathvik-Rao** [comment]
In same terminal
```
> npx @react-native-community/cli init sample
> cd sample
> npx react-native run-android
> taskkill /f /im node.exe
> npx react-native run-android
```

**Comment by nasunnguyenhuynh** [comment]
This is my way to resolve this stuff:
1. remove node_modules and package-lock.json, after that input in terminal `npm i`
2. `npx react-native doctor` and press 'f' to fix all problems
3. At your project `cd android` and `./gradlew clean assembleDebug`
4. `npm start` and press `a` for android

Env:
React-native 0.75.3
node v20.11.0
gradle 8.8
jdk-17.0.11
Android Studio Iguana | 2023.2.1 Patch 1
Build #AI-232.10300.40.2321.11567975, built on March 13, 2024
Runtime version: 17.0.9+0--11185874 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
Windows 11.0
GC: G1 Young Generation, G1 Old Generation
Memory: 2048M
Cores: 16
Registry:
    ide.experimental.ui=true
![image](https://github.com/user-attachments/assets/c4248475-c8ad-418d-9414-2befb06a6b1f)

**Comment by raulzc3** [comment]
> I was upgrading from 0.74.2 to 0.75.2 and got the same error on Android, here is what worked for me (using gradle-8.8-all.zip):
> 
> * remove node_modules, yarn.lock and android/.gradle
> * yarn install
> * cd android && ./gradlew --stop
> * ./gradlew clean
> * npx react-native start --reset-cache in the root of the project
> 
> Hope this will be helpful

This worked for me but with a few changes.
1. remove node_modules, android/.gradle and %userprofile%/.gradle (\.gradle if you use Linux)
2. yarn or npm install
3.  cd android && ./gradlew --stop
4. IMPORTANT: Close all terminals in and out of your editor (I use VSCode)
5. Open a new terminal (not in vscode) in your prject and run cd android && ./gradlew  clean
6. npx react-native start --reset-cache in the root of the project

This is the only way it works for me

**Comment by lalithasj** [comment]
Hi @nasunnguyenhuynh and @raulzc3 ,

Thank you for your suggestions, unfortunately, I didn't succeed. Please check the following errors :

TestSept19\android> ./gradlew clean assembleDebug
> Task :app:extractDebugVersionControlInfo FAILED

FAILURE: Build failed with an exception.           

* What went wrong:
Execution failed for task ':app:extractDebugVersionControlInfo'.                                      
> When VCS tagging is enabled (which is by default in release builds), the root project must be initialized with Git. Currently, Git is the only supported VCS for this feature.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.

BUILD FAILED in 1m 55s
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.
BUILD FAILED in 30s
error Failed to install the app. Command failed with exit code 1: gradlew.bat app:installDebug -PreactNativeDevServerPort=8081 FAILURE: Build failed with an exception. * What went wrong: Execution failed for task ':app:extractDebugVersionControlInfo'. > When VCS tagging is enabled (which is by default in release builds), the root project must be initialized with Git. Currently, Git is the only supported VCS for this feature. * Try: > Run with --stacktrace option to get the stack trace. > Run with --info or --debug option to get more log output. > Run with --scan to get full insights. > Get more he


./gradlew clean assembleDebug

FAILURE: Build failed with an exception.

* Where:
Settings file 'E:\ReactNative\TestSept19\android\settings.gradle' line: 2

* What went wrong:
Error resolving plugin [id: 'com.facebook.react.settings']
> java.io.UncheckedIOException: Could not move temporary workspace (E:\ReactNative\TestSept19\android\.gradle\8.8\dependencies-accessors\569c8b261a8a714d7731d5f568e0e5c05babae10-b080da13-52c7-4add-90b8-6b0226b7a474) to immutable location (E:\ReactNative\TestSept19\android\.gradle\8.8\dependencies-accessors\569c8b261a8a714d7731d5f568e0e5c05babae10)

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.

BUILD FAILED in 1s


npx react-native info
info Fetching system and libraries information...
System:
  OS: Windows 11 10.0.22631
  CPU: (8) x64 Intel(R) Core(TM) i5-10210U CPU @ 1.60GHz
  Memory: 6.25 GB / 15.85 GB
Binaries:
  Node:
    version: 20.17.0
    path: C:\Program Files\nodejs\node.EXE
  Yarn: Not Found
  npm:
    version: 9.6.2
    path: C:\Program Files\nodejs\npm.CMD
  Watchman: Not Found
SDKs:
  Android SDK:
    API Levels:
      - "28"
      - "29"
      - "30"
      - "31"
      - "32"
      - "33"
      - "33"
      - "33"
      - "34"
      - "35"
    Build Tools:
      - 28.0.3
      - 29.0.2
      - 29.0.3
      - 32.0.0
      - 33.0.3
      - 34.0.0
      - 35.0.0
    System Images:
      - android-28 | Google APIs Intel x86_64 Atom
      - android-29 | Google APIs Intel x86 Atom
      - android-30 | Google APIs Intel x86 Atom
      - android-31 | Google APIs Intel x86_64 Atom
      - android-34 | ARM 64 v8a
      - android-34 | Intel x86_64 Atom
    wanted: 0.75.3
  react-native-windows: Not Found
npmGlobalPackages:
  "*react-native*": Not Found
Android:
  hermesEnabled: true
  newArchEnabled: false
iOS:
  hermesEnabled: Not found
  newArchEnabled: Not found

./gradlew clean
Starting a Gradle Daemon, 3 stopped Daemons could not be reused, use --status for details

FAILURE: Build failed with an exception.

* Where:
Settings file 'E:\ReactNative\TestSept19\android\settings.gradle' line: 2

* What went wrong:
Error resolving plugin [id: 'com.facebook.react.settings']
> Included build 'E:\ReactNative\TestSept19\node_modules\@react-native\gradle-plugin' does not exist.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.

BUILD FAILED in 5s

and my final problem

npx react-native run-android
info Launching emulator...
error Failed to launch emulator. Reason: It took too long to start and connect with Android emulator: Pixel_7_API_34. You can try starting the emulator manually from the terminal with: C:\Users\Admin\AppData\Local\Android\Sdk/emulator/emulator @Pixel_7_API_34.
warn Please launch an emulator manually or connect a device. Otherwise app may fail to launch.
info Installing the app...

> Task :app:installDebug FAILED
Skipping device 'emulator-5554' (emulator-5554): Device is OFFLINE.
54 actionable tasks: 5 executed, 49 up-to-date

info 💡 Tip: Make sure that you have set up your development environment correctly, by running npx react-native doctor. To read more about doctor command visit: https://github.com/react-native-community/cli/blob/main/packages/cli-doctor/README.md#doctor 


FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:installDebug'.
> com.android.builder.testing.api.DeviceException: No online devices found.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.

BUILD FAILED in 2s
error Failed to install the app. Command failed with exit code 1: gradlew.bat app:installDebug -PreactNativeDevServerPort=8081 FAILURE: Build failed with an exception. * What went wrong: Execution failed for task ':app:installDebug'. > com.android.builder.testing.api.DeviceException: No online devices found. * Try: > Run with --stacktrace option to get the stack trace. > Run with --info or --debug option to get more log output. > Run with --scan to get full insights. > Get more help at https://help.gradle.org. BUILD FAILED in 2s.
info Run CLI with --verbose flag for more details.

I tried cd android> ./gradlew clean and npx react-native start --reset-cache 
But I didn't find any solution.

Appreciate any suggestions

Thanks

**Comment by Sathvik-Rao** [comment]
On Windows, I was able to get it working but with few issues:

1. Open the terminal and create a project:
   `> npx @react-native-community/cli init sample`
   `> cd sample`

2. Install any required npm packages.

3. Run the app:

   `sample> npx react-native run-android`

4. If you encounter an error, follow these steps:

   -  Kill any existing Node.js processes:

       `sample> taskkill /f /im node.exe`

   - Rerun the app:

       `sample> npx react-native run-android`

This approach will reload the app correctly. However, if you install new npm packages during development it fails to run again, this method may only work for fresh installations.

**Comment by lalithasj** [comment]
Hi @Sathvik-Rao 

I appreciate your suggestion. Unfortunately, I couldn't succeed. It does not recognize running android emulator.

taskkill /f /im node.exe
SUCCESS: The process "node.exe" with PID 20308 has been terminated.
SUCCESS: The process "node.exe" with PID 3152 has been terminated.
SUCCESS: The process "node.exe" with PID 19880 has been terminated.
SUCCESS: The process "node.exe" with PID 14196 has been terminated.
SUCCESS: The process "node.exe" with PID 4324 has been terminated.
SUCCESS: The process "node.exe" with PID 17528 has been terminated.
SUCCESS: The process "node.exe" with PID 9956 has been terminated.

TestSept19> npx react-native run-android
info Launching emulator...
error Failed to launch emulator. Reason: The emulator (Pixel_7_API_34) quit before it finished opening. You can try starting the emulator manually from the terminal with: C:\Users\Admin\AppData\Local\Android\Sdk/emulator/emulator @Pixel_7_API_34.
warn Please launch an emulator manually or connect a device. Otherwise app may fail to launch.
info Installing the app...

> Task :app:installDebug FAILED
Skipping device 'emulator-5554' (emulator-5554): Device is OFFLINE.
54 actionable tasks: 5 executed, 49 up-to-date

info 💡 Tip: Make sure that you have set up your development environment correctly, by running npx react-native doctor. To read more about doctor command visit: https://github.com/react-native-community/cli/blob/main/packages/cli-doctor/README.md#doctor 


FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:installDebug'.
> com.android.builder.testing.api.DeviceException: No online devices found.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.

BUILD FAILED in 3s
error Failed to install the app. Command failed with exit code 1: gradlew.bat app:installDebug -PreactNativeDevServerPort=8081 FAILURE: Build failed with an exception. * What went wrong: Execution failed for task ':app:installDebug'. > com.android.builder.testing.api.DeviceException: No online devices found. * Try: > Run with --stacktrace option to get the stack trace. > Run with --info or --debug option to get more log output. > Run with --scan to get full insights. > Get more help at https://help.gradle.org. BUILD FAILED in 3s.
info Run CLI with --verbose flag for more details.

Appreciate any suggestions.

Thanks

**Comment by Sathvik-Rao** [comment]
In my previous comment, I would like to add that during step 3 (`sample> npx react-native run-android`), I wait until the emulator is up and keep all terminals running without closing any. However, if I encounter any errors in the main terminal (`sample>`), I proceed to step 4.

If you're having trouble starting the emulator, it likely means the environment path variable for the SDK needs to be set up. Make sure to include `C:\Users\<your-user-name>\AppData\Local\Android\Sdk\platform-tools` in your PATH, and ensure that `JAVA_HOME` is correctly configured.

**Comment by lalithasj** [comment]
Hi @Sathvik-Rao 
Thank you for your suggestion, android emulator and metro are running properly. I did steps 3 and 4, but unfortunately, there is no progress.

**Comment by lalithasj** [comment]
Hi,
Thanks, everyone for your valuable suggestions. Upgrading Gradle, Android Studio, and Emulator helps to resolve problems.

**Comment by Agent009** [comment]
> Hey all, I've just tried to create a new project on 0.75.3 on Windows and I was able to successfully build with my machine. I also noticed several users in this thread managed to unblock themselves.
> 
> This issue seems to be related to Gradle > 8.6 versions and a new execution model which is not Windows-friendly (specifically if you have an Antivirus).
> 
> * ["Immutable workspace contents have been modified" failure when workspace contents are not modified gradle/gradle#28475](https://github.com/gradle/gradle/issues/28475)
> 
> We'll keep you updated if we have an official workaround other than the one we already shared here: [#46210 (comment)](https://github.com/facebook/react-native/issues/46210#issuecomment-2315210922)

I've tried all the permutations of the "fixes" posted so far, clearing all the necessary folders before doing so on each iteration, and none of them have resolved the issue for me.

It should be noted that I'm having issues with an existing project. A lot of the "fixes" seem to be for newly initiated projects.
However, clearing the cache, build and gradle folders should put existing projects in the same state as new projects so I'm not quite sure why it's working for some and not for others.

**Comment by cortinico** [comment]
Small update: I've been discussing with engineers on Gradle to attempt to mitigate this issue happening on Windows machines:
- https://github.com/gradle/gradle/pull/30681
- https://github.com/gradle/gradle/issues/30680

I will share more updates or workarounds as soon as we have them.

**Comment by vedang012** [comment]
i lowered react native version to 0.74.3 and gradle version to 8.5 and i worked for me!!

**Comment by BarbaraRamalho** [comment]
Do you have react-native installed? The new "Get Started Without a Framework" from React-Native's website asks you to uninstall previous react-native versions, but then never gives you the command to install them.

If you are just following the commands written you might fail to do so.

[](https://reactnative.dev/docs/getting-started-without-a-framework)

**Comment by twinedo** [comment]
For React Native 0.75.3, Please try :
- java version : 21
- gradle-wrapper.properties:  gradle-8.7-bin.zip
- gradle clean


make sure you're connected to internet

**Comment by Gagan1525** [comment]
> @Agent009 may be downgrade a version not on the same app but create a completely new project using `npx @react-native-community/cli init ProjectName --version=0.74`. just give it a try

Thanks, it worked for me

**Comment by ufukincibas** [comment]
Remove the id("com.facebook.react.settings") line from settings.gradle.

Keep the includeBuild('../node_modules/@react-native/gradle-plugin') line only within the pluginManagement block.

Clear caches and reinstall dependencies:
cd android && ./gradlew clean
npm cache clean --force
rm -rf node_modules/ && npm install

As a result, you can simplify your settings.gradle like this:

pluginManagement {
   includeBuild("../node_modules/@react-native/gradle-plugin")
}
rootProject.name = 'MyApp'
include ':app'


I hope these steps work and you can run your project successfully!

**Comment by alexsvt2** [comment]
> > @Agent009 may be downgrade a version not on the same app but create a completely new project using `npx @react-native-community/cli init ProjectName --version=0.74`. just give it a try
> 
> Thanks, it worked for me

This worked for me too but with a small fix, downgrade gradle from 8.6 to 8.4 in

`
`

**Comment by EzioLi01** [comment]
Using `npx react-native run-android` instead of `npm run android` is working for me.

**Comment by 10A0A** [comment]
> This is my way to resolve this stuff:
> 
> 1. remove node_modules and package-lock.json, after that input in terminal `npm i`
> 2. `npx react-native doctor` and press 'f' to fix all problems
> 3. At your project `cd android` and `./gradlew clean assembleDebug`
> 4. `npm start` and press `a` for android
> 
> Env: React-native 0.75.3 node v20.11.0 gradle 8.8 jdk-17.0.11 Android Studio Iguana | 2023.2.1 Patch 1 Build #AI-232.10300.40.2321.11567975, built on March 13, 2024 Runtime version: 17.0.9+0--11185874 amd64 VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o. Windows 11.0 GC: G1 Young Generation, G1 Old Generation Memory: 2048M Cores: 16 Registry: ide.experimental.ui=true ![image](https://private-user-images.githubusercontent.com/73988376/368839446-c4248475-c8ad-418d-9414-2befb06a6b1f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjg1NjYxODMsIm5iZiI6MTcyODU2NTg4MywicGF0aCI6Ii83Mzk4ODM3Ni8zNjg4Mzk0NDYtYzQyNDg0NzUtYzhhZC00MThkLTk0MTQtMmJlZmIwNmE2YjFmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDEwVDEzMTEyM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTkzN2U5YWVhYTg4OGJmZjZiNWJiZjhmZTY5YmQ4NjNhNjhjMjhhY2QzODRkMWExNjQ2NzhlNGU4YzE4M2E2YWMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.S4cJAorXDB1x5CaAQbeHVmGioVA8_P3K32f03jJoMFw)

Thanks, it works for me.

**Comment by sncbtz** [comment]
> > facing same issue
> 
> @rghanks
> 
> I've found a trick though. hit `npm run android`, let it do its job all the way. Now exit the terminal, clear the console and then again hit `npm run start`. Work for me.

Yeah it did the trick for me also 👍

**Comment by lf7817** [comment]
close windows defender

**Comment by nyagami** [comment]
Hi guys, as @cortinico stated, Windows Defender might be causing the issue. 
Follow [these steps](https://support.microsoft.com/en-us/windows/add-an-exclusion-to-windows-security-811816c0-4dfd-af4a-47e4-c301afe13b26#:~:text=Go%20to%20Start%20%3E%20Settings%20%3E%20Update,%2C%20file%20types%2C%20or%20process.) to ignore your project folder from Windows  Defender.

To @cortinico, I think RN should ask user to ignore project build from Windows Defenders when we run `npm run android` (like Android Studio asks when we start new android project)

Good luck!

**Comment by cortinico** [comment]
> Hi guys, as @cortinico stated, Windows Defender might be causing the issue. Follow [these steps](https://support.microsoft.com/en-us/windows/add-an-exclusion-to-windows-security-811816c0-4dfd-af4a-47e4-c301afe13b26#:~:text=Go%20to%20Start%20%3E%20Settings%20%3E%20Update,%2C%20file%20types%2C%20or%20process.) to ignore your project folder from Windows Defender.
> 
> To @cortinico, I think RN should ask user to ignore project build from Windows Defenders when we run `npm run android` (like Android Studio asks when we start new android project)

As I shared in my previous comment, we're waiting for a fix from Gradle to this matter.
A patch for this problem will ship in Gradle 8.11 which is not out yet.

It's true that disabling Windows Defender and other antivirus are likely to reduce this issue, due to how file locking works on Windows. Anyway we won't be officially recommending it as it exposes your system to security issues.

**Comment by nirmanyu007** [comment]
> Downgrading to RN `0.74.5` and Gradle `8.5` solves the issue. Wonder how the new release has been working for other folks. May just be an issue with those with simulated devices running on emulators?

thankyou so much brother because of you i was able to solve this error. I am so greatfull

**Comment by dubiao** [comment]
run
> cd android && ./gradlew --stop

and then rebuild. That saved my lifetime

**Comment by samarjeetmohanty** [comment]
**It worked for me** 
cd android
./gradlew clean

**Comment by ashikgowdaa** [comment]
I  used  **npx @react-native-community/cli@latest init MyApp** command to create native app , then i encountered an error while building
**java.io.UncheckedIOException: Could not move temporary workspace**,   the issue got solved using 

**npx react-native run-android**  this command , before running this clean gradle build .**/gradlew clean**  , 

distributionUrl=https\://services.gradle.org/distributions/gradle-8.7-all.zip

**Comment by Ydenizd** [comment]
> > > @Agent009 may be downgrade a version not on the same app but create a completely new project using `npx @react-native-community/cli init ProjectName --version=0.74`. just give it a try
> > 
> > 
> > Thanks, it worked for me
> 
> This also worked for me, but with a small fix: downgrading Gradle from 8.6 to 8.5.
> 
> android/gradle/wrapper/gradle-wrapper.properties `distributionUrl=https\://services.gradle.org/distributions/gradle-8.5-all.zip`

This one worked for me. Thanks

**Comment by bartlian** [comment]
it worked for me by yarn

**Comment by jjoaovitor7** [comment]
I was with this problem when upgrading from 0.75.2 to 0.76.2, and none of these fixed for me, but I was able to fix with:
- using openjdk 17.0.2, Android gradle plugin 8.6.0 (android/build.gradle) and Gradle 8.10.2 (`gradle-wrapper.properties`);
- changing in babel.config.js from `module:metro-react-native-babel-preset` to `module:@react-native/babel-preset`;
- changing `ndkVersion` to "27.2.12479018", `kotlin_version` to "2.0.0" and `buildToolsVersion` to "35.0.0" in `android/build.gradle`;
- migrating all `*.java` files to `*.kt` files and adding `apply plugin: "org.jetbrains.kotlin.android"` in `android/app/build`;
- update all deps and add the libs `react-native-community/cli` and `@react-native-community/cli-platform-android`;
- and finally running `gradlew clean`, `gradlew build` and `gradlew assembleRelease`.

**Comment by jjoaovitor7** [comment]
I was with this problem when upgrading from 0.75.2 to 0.76.2, and none of these fixed for me, but I was able to fix with:
- using openjdk 17.0.2, Android gradle plugin 8.6.0 (android/build.gradle) and Gradle 8.10.2 (`gradle-wrapper.properties`);
- changing in babel.config.js from `module:metro-react-native-babel-preset` to `module:@react-native/babel-preset`;
- changing `ndkVersion` to "27.2.12479018", `kotlin_version` to "2.0.0" and `buildToolsVersion` to "35.0.0" in `android/build.gradle`;
- migrating all `*.java` files to `*.kt` files and adding `apply plugin: "org.jetbrains.kotlin.android"` in `android/app/build`;
- update all deps and add the libs `react-native-community/cli` and `@react-native-community/cli-platform-android`;
- and finally running `gradlew clean`, `gradlew build` and `gradlew assembleRelease`.

**Comment by wufeng110226** [comment]
I have a better way, It seems to be caused by folder,
'Could not move temporary workspace (abcxxx-xxx-xxx) to immutable location (abc)'. So lets change it.
It worked

**Comment by wufeng110226** [comment]
I have a better way, It seems to be caused by folder,
'Could not move temporary workspace (abcxxx-xxx-xxx) to immutable location (abc)'. So lets change it.
It worked

**Comment by alexu96** [comment]
Hi Team, I tried all possible ways to fix this based on above comments but unfortunately nothing worked ! 
Requesting you to fix this asap !

**Comment by niyon-x** [comment]
> Hi Team, I tried all possible ways to fix this based on above comments but unfortunately nothing worked ! This issue exists only in Windows OS. I verified with Mac OS & it's working fine. Requesting team to fix this asap !
> 
> cc- @cortinico

 nah issue exist in macos as well. m3 pro, sequoia 15.1

**Comment by alexu96** [comment]
> > Hi Team, I tried all possible ways to fix this based on above comments but unfortunately nothing worked ! This issue exists only in Windows OS. I verified with Mac OS & it's working fine. Requesting team to fix this asap !
> > cc- @cortinico
> 
> nah issue exist in macos as well. m3 pro, sequoia 15.1

Mine is M1 pro, Sequoia 15.0

**Comment by lihappy** [comment]
In my case, I stopped the "yarn start" command in terminal, and gradle sync again in Android Studio, the errors gone.

**Comment by cortinico** [comment]
Just a heads up that Gradle 8.11.1 got released which should mitigate this issue.

If you're having this problem could you try to upgrade Gradle version with:

```
cd android && .\gradlew.bat wrapper --gradle-version=8.11.1 --distribution-type=all
```

After that, can you report back if the issue persists?

**Comment by El-mou33** [comment]
I had a similar issue: "Could not find com.facebook.react:react-native-gradle-plugin:0.71.0." It turned out to be caused by Java extensions in VSCode.

Try uninstalling the Java extensions in VSCode and reloading. You might need to do a clean install, but this worked for me. 

I know it isn't the exact issue but maybe that helps.

**Comment by nithronium** [comment]
> Just a heads up that Gradle 8.11.1 got released which should mitigate this issue.
> 
> If you're having this problem could you try to upgrade Gradle version with:
> 
> ```
> cd android
> ./gradlew wrapper --gradle-version=8.11.1 --distribution-type=all
> ./gradlew clean
> ```
> 
> After that, can you report back if the issue persists?

After spending many hours on debugging a lot of different stuff, I can confirm that this solved the problem for me. Thank you @cortinico

**Comment by vedantb2** [comment]
I just restarted vscode and it seems to have fixed it

**Comment by nhatduong-agilityio** [comment]
> I was upgrading from 0.74.2 to 0.75.2 and got the same error on Android, here is what worked for me (using gradle-8.8-all.zip):
> 
> * remove node_modules, yarn.lock and android/.gradle
> * yarn install
> * cd android && ./gradlew --stop
> * ./gradlew clean
> * npx react-native start --reset-cache in the root of the project
> 
> Hope this will be helpful

Does anyone have problems running ./gradlew clean?
<img width="831" alt="Screenshot_84" src="https://github.com/user-attachments/assets/46a39b9b-6809-4bdc-a494-b34a5c07082c" />

**Comment by cipolleschi** [comment]
@nhatduong-agilityio have you run `yarn install`? the error message says that the path `<your_app>\node_modules\@react-native\gradle-plugin` does not exists. This usually happens because you have not run `yarn install` and therefore there is no `node_modules` folder materialized

**Comment by nhatduong-agilityio** [comment]
@cipolleschi I have handled the problem. I install more gradle-plugin then delete package lock, delete node_module, then yarn install again, and run command ./gradlew clean before run react-native android. Thanks

**Comment by azmatalism** [comment]
> > facing same issue
> 
> @rghanks
> 
> I've found a trick though. hit `npm run android`, let it do its job all the way. Now exit the terminal, clear the console and then again hit `npm run start`. Work for me.

Work for me

**Comment by jayesh-vyas** [comment]
This error occurred when i was using `npx react-native run-android` but it worked when i used `npm run android`

**Comment by KHadham** [comment]
the weird thing is ,after all unworking solution out there 
restarting my machine solve this problem
```
android/settings.gradle' line: 4

* What went wrong:
Error resolving plugin [id: 'com.facebook.react.settings']
> Could not read workspace metadata from
```

**Comment by meshboy88** [comment]
I Found a solution. I'm using RN CLI: 0.76.5

0) Open the folder you want your app at VS Code
1) create app with: npx @react-native-community/cli init nameofyourapp
2) Run your emulator. Example: emulator -avd Pixel_4a_API_35
3)  First build MUST be with android and not with start. Example: yarn android.

This is working consistently for me.

**Comment by DewantiV** [comment]
just delete node_modules folder = > npm install => npm start -- --reset-cache => npm run android => done

**Comment by cortinico** [comment]
Closing as React Native 0.76.6 was released which contains a bump of Gradle to 8.11.1 which solves this issue for good.

**Comment by codewitharun** [comment]
I installed 0.76.6 today but the issue is still there

**Comment by ikarofelix** [comment]
I have 3 different projects with expo and I've been having a lot of similar problems lately, I've used 0.76.6, 0.75.2, 0.74... I've tried so many things and I'd would always have some problem, either this one, related to cache, sometimes it would work, the other day with the same config it would not.

I've managed to get it working by taking a look at /android/gradle/gradle-wrapper.properties, on the variable distributionUrl you have the gradle version. Go to https://docs.gradle.org/current/userguide/compatibility.html and take a look at the compatible java version, mine was 8.10.2, so I needed java 23, which I installed:
❯ java --version
openjdk 23 2024-09-17
OpenJDK Runtime Environment (build 23)
OpenJDK 64-Bit Server VM (build 23, mixed mode, sharing)

Now its working with just yarn android, without having to gradlew clean or even assemble as I have been doing a couple days ago. Don't forget to restart the terminal once you change the java version.

This could work for someone else, because yesterday I tried the same thing and it did not work for some strange reason. The version of react-native that people are saying 0.76.6 that has an update, did not work for me, so just give this a shot.

My dependencies versions, I'm also using yarn btw, not npm:

"@expo/config-plugins": "^9.0.14",
"@expo/prebuild-config": "^8.0.25",
"expo": "^52.0.17",
"expo-dev-client": "~5.0.5",
"react-native": "0.76.5",

**Comment by uralaltan** [comment]
For me moving the entire project to D:\ fixed it

**Comment by jdarshan5** [comment]
Running CMD with administrator worked for me
PS: I don't have a single idea why this is working, following a developer rule "If it works don't touch it 🫣"

**Comment by vweb69** [comment]
I have tried your provided solutions but these are not working in my condition isn't working 
here is settings. gradle

![Image](https://github.com/user-attachments/assets/07e5a55b-5af9-4138-a38e-07deae8c8a72)

here is gradle.wrapper.properties
![Image](https://github.com/user-attachments/assets/de2a0bc6-32d5-4027-8a39-d1daf3eac476)

**Comment by yamithell** [comment]
My build gave out many different errors on an empty project
It worked for me:

jdk-20
react-native 0.76.6
gradle 8.9

**Comment by NicholasKoldys** [comment]
For others experiencing the issue for fresh-latest builds, check the gradle wrapper version under 'distributionUrl' in "./_MyApp/android/gradle/wrapper/gradle-wrapper.properties".

Here my* gradle version was 8.10.2 via fresh RN0.77 install, (this was not working for me, through npm install/start/run)
for RN0.77 it should be 8.11.1 noted above (https://github.com/facebook/react-native/issues/46210#issuecomment-2586892665).

Also for noteworthy: pnpm run will give the same error with the updated gradlew, so using yarn at the moment.

**Comment by Prafullsingh975** [comment]
I was also facing the same problem. My react-native version was **0.75.13**. I upgraded it to react-native version **0.77.0** and resolved my issue.

Steps:
- go to https://react-native-community.github.io/upgrade-helper/
Please make the changes that are mentioned.
- restart the pc
- go to C:\Users\User_Name\.gradle
- delete caches and wrapper folder
- delete node_module and package-log.json
- do npm i 
- delete debug build
- run npx react-native run-android
# and my problem was solved

**Comment by SorenHK** [comment]
> I've managed to get it working by taking a look at /android/gradle/gradle-wrapper.properties, on the variable distributionUrl you have the gradle version. Go to https://docs.gradle.org/current/userguide/compatibility.html and take a look at the compatible java version, mine was 8.10.2, so I needed java 23, which I installed: ❯ java --version openjdk 23 2024-09-17 OpenJDK Runtime Environment (build 23) OpenJDK 64-Bit Server VM (build 23, mixed mode, sharing)

Thank you! The solution for me ended up being just updating to JDK 23 like you mentioned. I am on Gradle 8.10.2 and React Native 0.74.7

**Comment by rampazzo1989** [comment]
I've got it working for me by running gradlew clean command with sudo. Yeah. I know.

**Comment by yashseth391** [comment]
@cwoolum 

If you have researched it , you could have find you can run ./gradlew.bat clean

**Comment by mauseoluwasegun** [comment]
@cwoolum please help me out:
./gradlew.bat clean

FAILURE: Build failed with an exception.

* Where:
Settings file 'C:\Users\USER\Desktop\project\podcastcrest\podcast_client\android\settings.gradle' line: 2

* What went wrong:
Error resolving plugin [id: 'com.facebook.react.settings']
> A problem occurred configuring project ':gradle-plugin'.
   > A build operation failed.
         Could not move temporary workspace (C:\Users\USER\.gradle\caches\8.11.1\transforms\2808fad2bc90d75494ed720eb50db5cb-36a70b58-de32-423c-a685-98399b1e6249) to immutable location (C:\Users\USER\.gradle\caches\8.11.1\transforms\2808fad2bc90d75494ed720eb50db5cb)
      > Could not move temporary workspace (C:\Users\USER\.gradle\caches\8.11.1\transforms\2808fad2bc90d75494ed720eb50db5cb-36a70b58-de32-423c-a685-98399b1e6249) to immutable location (C:\Users\USER\.gradle\caches\8.11.1\transforms\2808fad2bc90d75494ed720eb50db5cb)

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.

BUILD FAILED in 7s

**Comment by sanaf1333** [comment]
I am facing same issue, have you found any solution?

> [@cwoolum](https://github.com/cwoolum) please help me out: ./gradlew.bat clean
> 
> FAILURE: Build failed with an exception.
> 
> * Where:
>   Settings file 'C:\Users\USER\Desktop\project\podcastcrest\podcast_client\android\settings.gradle' line: 2
> * What went wrong:
>   Error resolving plugin [id: 'com.facebook.react.settings']
> 
> > A problem occurred configuring project ':gradle-plugin'.
> > A build operation failed.
> > Could not move temporary workspace (C:\Users\USER.gradle\caches\8.11.1\transforms\2808fad2bc90d75494ed720eb50db5cb-36a70b58-de32-423c-a685-98399b1e6249) to immutable location (C:\Users\USER.gradle\caches\8.11.1\transforms\2808fad2bc90d75494ed720eb50db5cb)
> > > Could not move temporary workspace (C:\Users\USER.gradle\caches\8.11.1\transforms\2808fad2bc90d75494ed720eb50db5cb-36a70b58-de32-423c-a685-98399b1e6249) to immutable location (C:\Users\USER.gradle\caches\8.11.1\transforms\2808fad2bc90d75494ed720eb50db5cb)
> 
> * Try:
> 
> > Run with --stacktrace option to get the stack trace.
> > Run with --info or --debug option to get more log output.
> > Run with --scan to get full insights.
> > Get more help at https://help.gradle.org.
> 
> BUILD FAILED in 7s

I am facing same issue, have you found any solution?

**Comment by KruzTheMarvelHero** [comment]
info A dev server is already running for this project on port 8081.
info Installing the app...

info 💡 Tip: Make sure that you have set up your development environment correctly, by running npx react-native doctor. To read more about doctor command visit: https://github.com/react-native-community/cli/blob/main/packages/cli-doctor/README.md#doctor 


FAILURE: Build failed with an exception.

* Where:
Settings file '/Volumes/DATA-D/ReactNative/ICC/SourceCode/IndiaCustomerCare/android/settings.gradle' line: 2

* What went wrong:
Error resolving plugin [id: 'com.facebook.react.settings']
> Could not read workspace metadata from /Users/krunal/.gradle/caches/8.13/kotlin-dsl/accessors/cbd2677dac5813b0a8ba9dfa391abe66/metadata.bin

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.

BUILD FAILED in 758ms
error Failed to install the app. Command failed with exit code 1: ./gradlew app:installDebug -PreactNativeDevServerPort=8081 FAILURE: Build failed with an exception. * Where:
Settings file '/Volumes/DATA-D/ReactNative/ICC/SourceCode/IndiaCustomerCare/android/settings.gradle' line: 2 * What went wrong:
Error resolving plugin [id: 'com.facebook.react.settings']
> Could not read workspace metadata from /Users/krunal/.gradle/caches/8.13/kotlin-dsl/accessors/cbd2677dac5813b0a8ba9dfa391abe66/metadata.bin * Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org. BUILD FAILED in 758ms.
info Run CLI with --verbose flag for more details.

Please share some suggestions, 

RN version : 0.79
distributionUrl=https\://services.gradle.org/distributions/gradle-8.13-bin.zip

build.gradle File

buildscript {
    ext {
        buildToolsVersion = "35.0.0"
        minSdkVersion = 24
        compileSdkVersion = 35
        targetSdkVersion = 35
        ndkVersion = "27.1.12297006"
        kotlinVersion = "2.0.21"
    }
    repositories {
        google()
        mavenCentral()
    }
    dependencies {
        classpath("com.android.tools.build:gradle")
        classpath("com.facebook.react:react-native-gradle-plugin")
        classpath("org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlinVersion")
    }
}

apply plugin: "com.facebook.react.rootproject"

**Comment by liangskyli** [comment]
react-native@0.75-0.80 all error in windows

**Comment by felipescola** [comment]
Did anyone manage to solve this issue:
What went wrong:
Error resolving plugin [id: 'com.facebook.react.settings']

I can run it normally on the emulator, but when I try to run ./gradlew assembleRelease, I get the error above. Running gradlew clean also causes an error.

**Comment by JVGS1111** [comment]
@felipescola did you see that?
https://github.com/facebook/react-native/issues/46046#issuecomment-2291641341
This solved my problem.
